pkg/webhooks/provisioning/v1alpha1/defaulter.go

```diff
@@ -49,20 +53,39 @@ func (v *Defaulter) Handle(ctx context.Context, req admission.Request) admission
 	if err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
-	v.applyDefaults(&provisioner.Spec)
+	if err := v.applyDefaults(ctx, &provisioner.Spec); err != nil {
+		return admission.Errored(0, fmt.Errorf("applying defaults to provisioner, %w", err))
+	}
+
 	marshaled, err := json.Marshal(provisioner)
 	if err != nil {
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaled)
 }
 
-func (v *Defaulter) applyDefaults(spec *provisioning.ProvisionerSpec) {
-	v.defaultTTL(spec)
+func (v *Defaulter) applyDefaults(ctx context.Context, spec *provisioning.ProvisionerSpec) error {
+	return functional.ValidateAll(
```

Comment:
If we use this in this way, we should rename that function to something else. It used to be AllSucceed/SucceedAll

Reply:
👍